src/test/groovy/com/stormpath/tck/authentication/CookieIT.groovy

```diff
@@ -0,0 +1,159 @@
+import static org.hamcrest.Matchers.*
+import static org.testng.Assert.*
+import static com.stormpath.tck.util.FrameworkConstants.MeRoute
+import static com.stormpath.tck.util.FrameworkConstants.OauthRoute
+import static com.stormpath.tck.util.CookieUtils.isCookieDeleted
+
+@Test
+class CookieIT extends AbstractIT {
+
+    def createTestAccountTokens() {
+        def account = new TestAccount()
+        account.registerOnServer()
+
+        def response =
+                given()
+                    .param("grant_type", "password")
+                    .param("username", account.username)
+                    .param("password", account.password)
+                .when()
+                    .post(OauthRoute)
+                .then()
+                    .statusCode(200)
+                    .contentType(ContentType.JSON)
+                    .body("access_token", not(isEmptyOrNullString()))
+                .extract()
+                    .response()
+        deleteOnClassTeardown(account.href)
+
+        return new Tuple2(response.path("access_token"), response.path("refresh_token"))
+    }
+
+    /** If access token is invalid, use refresh token to get new access token
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/34">#34</a>
+     * @throws Exception
+     */
+    @Test(groups=["v100"])
+    public void usesRefreshTokenToGetNewAccessToken() throws Exception {
+        def (String accessToken, String refreshToken) = createTestAccountTokens()
+
+        given()
+            .cookie("refresh_token", refreshToken)
+        .when()
+            .get(MeRoute)
+        .then()
+            .statusCode(200)
+            .cookie("access_token", not(accessToken))
+            .cookie("refresh_token", is(refreshToken))
+    }
+
+    /** If refresh token is invalid, reject request and delete cookies
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/35">#35</a>
+     * @throws Exception
+     */
+    @Test(groups=["v100"])
+    public void deleteCookiesIfRefreshTokenIsInvalid() throws Exception {
+        def response = given()
+            .cookie("refresh_token", "not_a_valid_refresh_token_at_all")
+        .when()
+            .get(MeRoute)
+        .then()
+            .statusCode(401)
+        .extract()
+            .response()
+
+        assertTrue(isCookieDeleted(response.detailedCookies.get("access_token")))
+        assertTrue(isCookieDeleted(response.detailedCookies.get("refresh_token")))
+    }
+
+    /** Reject unauthorized text/html requests with 302 to login route
+     * 302 to login route should preserve original URL including query string
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/36">#36</a>
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/121">#121</a>
+     * @throws Exception
+     */
+    @Test(groups=["v100"])
+    public void unauthorizedRequestIsForwardedToLogin() throws Exception {
```

Comment:
Actually, for `Accept: text/html` depending on how the endpoint is configured it could potentially do this.